pkg/builder/backup_builder.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	velerov1api "github.com/heptio/velero/pkg/apis/velero/v1"
+)
+
+/*
+
+Example usage:
+
+var backup = builder.ForBackup("velero", "backup-1").
+	ObjectMeta(
+		builder.WithLabels("foo", "bar"),
+		builder.WithClusterName("cluster-1"),
+	).
+	SnapshotVolumes(true).
+	Result()
+
+*/
+
+// BackupBuilder builds Backup objects.
+type BackupBuilder struct {
+	object *velerov1api.Backup
+}
+
+// ForBackup is the constructor for a BackupBuilder.
+func ForBackup(ns, name string) *BackupBuilder {
```

Comment:
just a thought: we could forgo the `For` prefix, `builder.Backup` still makes sense

Reply:
Personally I slightly prefer it with the `For` prefix, but I'm 100% fine with whatever the majority prefers - @nrb @carlisia?

Reply:
In this case, I agree with @skriss and prefer `ForBackup`. If the function was named `Backup`, I'd be left wondering why I'm getting back an object that is not a `Backup`. Adding the `For` answers that question for me.